File: test_api.js
import fs from 'fs';
import path from 'path';

let apiKey = process.env.GEMINI_API_KEY;
if (!apiKey && fs.existsSync('.env')) {
  const envContent = fs.readFileSync('.env', 'utf8');
  for (const line of envContent.split('\n')) {
    if (line.startsWith('GEMINI_API_KEY=')) {
      apiKey = line.split('=')[1].trim();
    }
  }
}

async function testModel(model) {
  const url = `https://generativelanguage.googleapis.com/v1beta/models/${model}:generateContent?key=${apiKey}`;
  const payload = {
    contents: [{ parts: [{ text: `Return JSON: {"status": "success", "model": "${model}"}` }] }],
    generationConfig: { responseMimeType: 'application/json' }
  };
  try {
    const res = await fetch(url, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(payload)
    });
    if (!res.ok) {
      const txt = await res.text();
      console.log(`[FAIL] ${model}: HTTP ${res.status} - ${txt}`);
      return false;
    }
    const data = await res.json();
    console.log(`[OK] ${model} works:`, data.candidates[0].content.parts[0].text);
    return true;
  } catch (err) {
    console.log(`[ERR] ${model}:`, err.message);
    return false;
  }
}

for (const m of ['gemini-2.5-flash', 'gemini-2.0-flash', 'gemini-1.5-flash']) {
  await testModel(m);
}

File: test_api.py
import os
import json
import urllib.request

api_key = os.environ.get('GEMINI_API_KEY')
if not api_key and os.path.exists('.env'):
    with open('.env', 'r', encoding='utf-8') as f:
        for line in f:
            if line.startswith('GEMINI_API_KEY='):
                api_key = line.split('=', 1)[1].strip()

# Testing gemini-2.5-flash / gemini-2.0-flash / gemini-1.5-flash
for model in ['gemini-2.5-flash', 'gemini-2.0-flash', 'gemini-1.5-flash']:
    url = f'https://generativelanguage.googleapis.com/v1beta/models/{model}:generateContent?key={api_key}'
    data = {
        'contents': [{'parts': [{'text': 'Return JSON: {"status": "success", "model": "' + model + '"}'}]}],
        'generationConfig': {'responseMimeType': 'application/json'}
    }
    req = urllib.request.Request(url, data=json.dumps(data).encode('utf-8'), headers={'Content-Type': 'application/json'})
    try:
        with urllib.request.urlopen(req) as resp:
            res = json.loads(resp.read().decode('utf-8'))
            print(f'✅ Model {model} succeeded:', res['candidates'][0]['content']['parts'][0]['text'].strip())
            break
    except Exception as e:
        print(f'❌ Model {model} failed:', str(e))
